remote-config/src/fetch/fetcher.rs

```diff
@@ -0,0 +1,826 @@
+                                    length: decoded.len() as i64,
+                                    hashes: target_file
+                                        .hashes
+                                        .iter()
+                                        .map(|(algorithm, hash)| TargetFileHash {
+                                            algorithm: algorithm.to_string(),
+                                            hash: hash.to_string(),
+                                        })
+                                        .collect(),
+                                },
+                                handle: if let Some(handle) = handle {
+                                    self.file_storage.update(&handle, version, decoded)?;
+                                    handle
+                                } else {
+                                    self.file_storage.store(version, parsed_path, decoded)?
+                                },
+                                expiring: false,
+                            },
+                        );
+                    } else {
+                        anyhow::bail!("Failed parsing version from remote config path {path}");
+                    }
+                } else {
+                    anyhow::bail!(
+                        "Failed base64 decoding config for path {path}: {}",
+                        String::from_utf8_lossy(raw_file)
+                    )
+                }
+            } else {
+                anyhow::bail!(
+                    "Found changed config data for path {path}, but no file; existing files: {:?}",
+                    incoming_files.keys().collect::<Vec<_>>()
+                )
+            }
+        }
+
+        let mut configs = Vec::with_capacity(config_paths.len());
+        for config in config_paths.iter() {
+            if let Some(target_file) = target_files.get_mut(config as &dyn RemoteConfigPathType) {
+                target_file.expiring = false;
+                configs.push(target_file.handle.clone());
+            } else {
+                anyhow::bail!("Found {config} in client_configs response, but it isn't stored.");
+            }
+        }
+
+        opaque_state.targets_version = targets_list.signed.version as u64;
+        opaque_state.last_configs = response.client_configs;
+        opaque_state.last_config_paths = config_paths;
+        Ok(Some(configs))
+    }
+}
+
+fn get_product_endpoint(subdomain: &str, endpoint: &Endpoint) -> Endpoint {
+    let mut parts = endpoint.url.clone().into_parts();
+    if endpoint.api_key.is_some() {
+        if parts.scheme.is_none() {
+            parts.scheme = Some(Scheme::HTTPS);
+            parts.authority = Some(
+                format!("{}.{}", subdomain, parts.authority.unwrap())
+                    .parse()
+                    .unwrap(),
+            );
+        }
+        parts.path_and_query = Some(PathAndQuery::from_static("/api/v0.1/configurations"));
```

Comment:
I don't think there's ever a use case where a tracer client should be talking to /api/v0.1/configurations as that's the RC backend. The RC backend is not designed to talk directly to tracers and only to what we call the "core remote config service" running in the agent.

Reply:
It's sort of unused, just makes the code do something valid at this place, the actual run loop has `if self.state.endpoint.api_key.is_some() { return Err() }`.

I don't intend to use it without thorough discussion with RC team.

Reply:
Whatever, removed.